pkg/internal/checkup/checkup.go

```diff
@@ -283,6 +287,31 @@ func newTrafficGeneratorPod(checkupConfig config.Config, secondaryNetworkRequest
 	)
 }
 
+func (c *Checkup) waitForPodDeletion(ctx context.Context) error {
+	if c.trafficGeneratorPod == nil {
+		return fmt.Errorf("failed to delete traffic generator Pod, object doesn't exist")
+	}
```

Comment:
If we've reached here, the `trafficGeneratorPod` is not nil.
IMO this `if` is redundant.

Reply:
it's more wholesome but you've got a point. removing